libs/ml/include/ml/ops/ops.hpp

```diff
@@ -149,6 +151,60 @@ class Ops
     // Empty deafult implementation for non-trainable Ops.
   }
 
+  /**
+   * @brief ChargeForward
+   * @return estimated charge amount, necessary for performing a forward pass on data of given
+   * shapes.
+   */
+  virtual MLChargeAmount ChargeForward()
+  {
+    // TODO(VH): make a pure virtual call;
+    FETCH_LOG_ERROR(Descriptor(),
+                    " Error: call to unexisting ChargeForward() implementation! returned 0.");
+    return 0;
+  }
+
+  /**
+   * @brief ChargeBackward
+   * @return estimated charge amount, necessary for performing a backward pass on data of given
+   * shapes.
+   */
+  virtual MLChargeAmount ChargeBackward() const
+  {
+    // TODO(VH): make a pure virtual call;
+    FETCH_LOG_ERROR(Descriptor(),
+                    " Error: call to unexisting ChargeBackward() implementation! returned 0.");
+    return 0;
+  }
+
+  /**
+   * @brief TotalElementsIn calculated a sum of total elements in all given tensors
+   * @param shapes - vector of tensor shapes
+   */
+  static fetch::math::SizeType TotalElementsIn(std::vector<fetch::math::SizeVector> const &shapes)
+  {
+    using fetch::math::SizeType;
+    if (shapes.empty())
+    {
+      return 0;
+    }
+    SizeType total_elements = 0;
+    for (fetch::math::SizeVector const &shape : shapes)
+    {
+      SizeType elements_in_shape = 1;
+      for (SizeType dimension : shape)
+      {
+        if (dimension == 0)
+        {
+          throw fetch::math::exceptions::WrongShape("A dimension of size 0 found in tensor shape!");
+        }
+        elements_in_shape *= dimension;
+      }
```

Comment:
i believe there is a function in math/matrix_operations that might do this for you called Product. Might as well use that since it's more succinct and could potentially be better optimised in the future

Reply:
nice suggestion! I'll use it then.